cmd/state/stateless/state.go

```diff
@@ -423,60 +498,40 @@ beginTx:
 			return err
 		}
 
-		skipFirst := len(r.HistoryKey) > 0 // snapshot stores last analyzed key
 		for k, vIsEmpty, err := c.SeekKey(r.HistoryKey); k != nil || err != nil; k, vIsEmpty, err = c.NextKey() {
 			if err != nil {
 				return err
 			}
-			if skipFirst {
-				skipFirst = false
-				continue
+			r.HistoryKey = k
+			i++
+			if r.interrupt(ctx, i, startTime) {
+				return nil
 			}
-
 			copy(addrHash[:], k[:32]) // First 20 bytes is the address
-			addr := string(addrHash.Bytes())
 			copy(hash[:], k[40:72])
 			timestamp, _ := dbutils.DecodeTimestamp(k[72:])
+
+			if timestamp > r.StartedWhenBlockNumber { // only count what happened before analysis started
+				continue
+			}
+
+			addr2HashKey := append(addrHash.Bytes(), hash.Bytes()...)
```

Comment:
Note that here you are effectively making a copy of addrHash in order to resize it, that is why it works for Storage history, but not for accounts history

Reply:
But perhaps it is better to have explicit copy